src/goto-programs/slice_global_inits.cpp

```diff
@@ -79,6 +77,38 @@ void slice_global_inits(goto_modelt &goto_model)
 
   goto_programt &goto_program=f_it->second.body;
 
+  // add all symbols from right-hand sides of required symbols
+  bool fixed_point_reached = false;
+  std::vector<bool> seen(goto_program.instructions.size(), false);
+  while(!fixed_point_reached)
+  {
+    fixed_point_reached = true;
+
+    std::vector<bool>::iterator seen_it = seen.begin();
+    forall_goto_program_instructions(i_it, goto_program)
+    {
+      if(!*seen_it && i_it->is_assign())
+      {
+        const code_assignt &code_assign = i_it->get_assign();
+        const irep_idt id = to_symbol_expr(code_assign.lhs()).get_identifier();
+
+        if(
+          has_prefix(id2string(id), CPROVER_PREFIX) ||
+          symbols.find(id) != symbols.end())
```

Comment:
Perhaps rename `symbols_to_keep` to make clearer what's happening here?

Reply:
Done.